packages/nodes-base/nodes/WhatsApp/WhatsApp.node.ts

```diff
@@ -50,6 +61,53 @@ export class WhatsApp implements INodeType {
 			...mediaFields,
 			...messageTypeFields,
 			...mediaTypeFields,
+			...getSendAndWaitProperties([], 'message', undefined, {
+				noButtonStyle: true,
+				defaultApproveLabel: '✓ Approve',
+				defaultDisapproveLabel: '✗ Decline',
+			}).filter((p) => p.name !== 'subject'),
 		],
 	};
+
+	webhook = sendAndWaitWebhook;
+
+	customOperations = {
+		message: {
+			async [SEND_AND_WAIT_OPERATION](this: IExecuteFunctions) {
```

Comment:
How is this different from doing `if (resource === "message" && operation === SEND_AND_WAIT_OPERATION) {...}` in the `execute` function?

Reply:
it's not, first [implementation](https://github.com/n8n-io/n8n/pull/12941/commits/ddea38d1babf50a12188ac24035a3cd2008bdcb2) used different `Interface` and code suppose to go inside `execute`

implementation was updated base of previous reviews

I do think this is better approach then hijack `execute` function for specific resource and operation, meaning for developer not familiar with routing nodes would be very confusing to see only one resource and operation in `execute` method